### PR TITLE
core: more logs in ManagedChannelImpl.

### LINKED ISSUE #2299: More logs in channel
We have insufficient logging in the channel layer. In particular:
- Name resolution success and failures. Although failures are plumbed to application when it makes a call.
- ManagedChannelImpl goes into and out of idleness

Whether they should be INFO or FINE are not clear, but at least they should start as FINE.

### PR DESCRIPTION
**Pull request by zhangkun83**
They have already been added to ManagedChannelImpl2.

Resolves #2299

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/ManagedChannelImpl.java [hunk 1]
```diff
@@ -764,10 +769,13 @@ public void onUpdate(List<ResolvedServerInfoGroup> servers, Attributes config) {
         onError(Status.UNAVAILABLE.withDescription("NameResolver returned an empty list"));
         return;
       }
+      log.log(Level.FINE, "[{0}] resolved address: {1}, config={2}",
+          new Object[] {getLogId(), servers, config});
 
       try {
         balancer.handleResolvedAddresses(servers, config);
       } catch (Throwable e) {
+        log.log(Level.WARNING, "[" + getLogId() + "] Caught exception from LoadBalancer", e);
```
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/internal/ManagedChannelImpl.java]: Why is this here?  shouldnt the balancer handle this below?

Comment (zhangkun83) [inline core/src/main/java/io/grpc/internal/ManagedChannelImpl.java]: So that we don't need to repeat the log in each LoadBalancer. Also we don't control 3rd-party LoadBalancers and cannot guarantee they will handle it or log it. This is for our own benefit, in that we always have a way to know what's happening.

Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/internal/ManagedChannelImpl.java]: Won't a lot of other alarm bells go off?  This is expected to happen for network outages in common case right?

Comment (zhangkun83) [inline core/src/main/java/io/grpc/internal/ManagedChannelImpl.java]: No, it's only logged if there is a bug in `LoadBalancer`, because it's not expected to throw.

Here we catch and pass the exception back to `LoadBalancer`, in hope that it will be propagated to the application by LoadBalancer. However, LoadBalancer may not propagate or log as we expect, or there may not even be RPC yet to be propagated to, so we need to log here as the last resort.

We don't catch in `onError()` below, because if `handleNameResolutionError()` throws, there isn't a really good way to propagate this error to application. The exception will be caught by transport thread (or `ChannelExecutor` in v2 impl) and logged there.

Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/internal/ManagedChannelImpl.java]: If this is unexpected, could you put that in the message?  This [issue](https://github.com/grpc/grpc-java/issues/2604) from yesterday came from someone not realizing they should have been catching the exception, and reading the log message.

Comment (zhangkun83) [inline core/src/main/java/io/grpc/internal/ManagedChannelImpl.java]: Done. PTAL.
